sale_comment_template/tests/test_sale_order_report.py

```diff
@@ -1,74 +1,76 @@
 # Copyright 2017 Simone Rubino - Agile Business Group
 # Copyright 2018 Tecnativa - Pedro M. Baeza
+# Copyright 2021 Tecnativa - Víctor Martínez
 # License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
 
-import odoo.tests
 from odoo.tests.common import TransactionCase
 
 
-@odoo.tests.common.at_install(False)
-@odoo.tests.common.post_install(True)
 class TestAccountInvoiceReport(TransactionCase):
-    def setUp(self, *args, **kwargs):
-        super(TestAccountInvoiceReport, self).setUp()
+    def setUp(self):
+        super().setUp()
+        self.company = self.env.ref("base.main_company")
         self.base_comment_model = self.env["base.comment.template"]
-        self.before_comment = self._create_comment("before_lines")
-        self.after_comment = self._create_comment("after_lines")
-        self.partner_id = self.env["res.partner"].create({"name": "Partner Test"})
-        self.sale_order = self.env.ref("sale.sale_order_7")
+        # Create comment related to sale model
+        self.sale_obj = self.env.ref("sale.model_sale_order")
+        self.sale_before_comment = self._create_comment(self.sale_obj, "before_lines")
+        self.sale_after_comment = self._create_comment(self.sale_obj, "after_lines")
+        # Create comment related to move model
+        self.move_obj = self.env.ref("account.model_account_move")
+        self.move_before_comment = self._create_comment(self.move_obj, "before_lines")
+        self.move_after_comment = self._create_comment(self.move_obj, "after_lines")
+        # Create partner
+        self.partner = self.env["res.partner"].create({"name": "Partner Test"})
+        self.partner.base_comment_template_ids = [
+            (4, self.sale_before_comment.id),
+            (4, self.sale_after_comment.id),
+            (4, self.move_before_comment.id),
+            (4, self.move_after_comment.id),
+        ]
+        self.sale_order = self.env.ref("sale.sale_order_7").copy(
+            {"partner_id": self.partner.id, "partner_invoice_id": self.partner.id}
+        )
```

Comment:
Can you create a specific sale order instead of copying? Or, if you copy, can you assign the proper company id? I think this might cause integration issues if not solved...

Reply:
Thanks, changes done.